src/libcore/lib.rs

```diff
@@ -50,6 +50,7 @@
 
 // Since libcore defines many fundamental lang items, all tests live in a
 // separate crate, libcoretest, to avoid bizarre issues.
+#![cfg(not(test))]
```

Comment:
This feels odd; presumably independent of whether we are running tests we would expect core to exist and define everything -- or am I misreading what this does?

Reply:
@Mark-Simulacrum 

The problem is when we are testing the library, we will build an executable and link to libtest. Now, this executable defines some lang items, and at the same time libtest imports the non-testing libcore which also defines some lang items. This will cause the E0152 "duplicate lang item found" error. I'm not sure why this is not discovered before, maybe no one tried to `./x.py test src/libcore`? 🙃 

Reduced test case:

```rust
#![cfg(not(test))] // <-- comment out to cause E0152 when `cargo test`.

#![feature(no_core, lang_items)]
#![no_core]

/// ```
///                               
/// ```
#[lang = "sized"]
pub trait Sized {}
```

In `cargo test`, the doc tests are still discovered and executed even with this cfg. I assume this is the same for rustbuild, though if it turns out to be wrong we could change the `cfg` to `any(dox, not(test))`.

I've updated this comment to explain the attribute.